File: core-geo/src/main/java/com/example/core_geo/Point.java
package com.example.core_geo;

import android.os.Parcelable;

import com.google.auto.value.AutoValue;

/**
 * Created by lev.novikov
 * Date: 18/12/17.
 */

@AutoValue
public abstract class Point implements Parcelable {

    public abstract Coordinates coordinates();

    public abstract String title();

    public static Point create(Coordinates coordinates, String title) {
        return new AutoValue_Point(coordinates, title);
    }
}

File: core-geo/src/main/java/com/example/core_geo/Point.kt
package com.example.core_geo

import android.annotation.SuppressLint
import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

/**
 * Created by lev.novikov
 * Date: 18/12/17.
 */

@SuppressLint("ParcelCreator")
@Parcelize
data class Point(val coordinates: Coordinates, val title: String) : Parcelable